### PR TITLE
Agda

### PR DESCRIPTION
**Pull request by Decodetalkers**
add agda support, with the best treesitter until now, until we find a better one

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:60]: wrong parser...

File: lua/nvim-treesitter/parsers.lua
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:53]: please ask them first whether they are OK with being listed as maintainer for _nvim-treesitter_!

Comment (Decodetalkers) [reply lua/nvim-treesitter/parsers.lua:53]: I have asked him..

Comment (clason) [reply lua/nvim-treesitter/parsers.lua:53]: And did they agree?

Comment (Decodetalkers) [reply lua/nvim-treesitter/parsers.lua:54]: I will wait for his reply

Comment (clason) [reply lua/nvim-treesitter/parsers.lua:54]: To be clear, this is just about listing them as the maintainer. The parser can be added even without a maintainer listed.

Comment (theHamsta) [reply lua/nvim-treesitter/parsers.lua:54]: Typically the person adding the parser cares about this language and becomes contact person for nvim-treesitter

**Review by theHamsta — COMMENTED** [review]
Looks good apart from the maintainer field that should be the person updating the files in Nvim-treesitter for This parser

File: lua/nvim-treesitter/parsers.lua
Comment (theHamsta) [reply lua/nvim-treesitter/parsers.lua:54]: @Decodetalkers friendly ping. Can you fix this?

Comment (Decodetalkers) [reply lua/nvim-treesitter/parsers.lua:54]: Sorry, fix what?

Comment (Decodetalkers) [reply lua/nvim-treesitter/parsers.lua:54]: > @Decodetalkers friendly ping. Can you fix this?

Sorry, fix what?

Comment (theHamsta) [reply lua/nvim-treesitter/parsers.lua:54]: to put you as maintainer instead of the person who wrote the parser

Comment (Decodetalkers) [reply lua/nvim-treesitter/parsers.lua:54]: I see

Comment (Decodetalkers) [reply lua/nvim-treesitter/parsers.lua]: > to put you as maintainer instead of the person who wrote the parser

Now changed..

File: queries/agda/highlights.scm
Comment (theHamsta) [inline queries/agda/highlights.scm:3]: isn't interger `@number`?

**Comment by theHamsta** [comment]
also rebasing is necessary

File: queries/agda/highlights.scm
Comment (Decodetalkers) [reply queries/agda/highlights.scm]: > isn't interger `@number`?

fixed

**Review by theHamsta — APPROVED** [review]
Thank you!

**Pull request merged**
